src/Workspaces/VisualBasic/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.vb

```diff
@@ -61,6 +61,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions.ContextQuery
         End Function
 
         <Extension()>
+        Public Function IsNamespaceDeclarationNameContext(syntaxTree As SyntaxTree, position As Integer, cancellationToken As CancellationToken) As Boolean
+            Dim statement = syntaxTree.GetTargetToken(position, cancellationToken).GetAncestor(Of NamespaceStatementSyntax)
+            Return statement IsNot Nothing AndAlso statement.Name.Span.IntersectsWith(position)
```

Comment:
1. Assuming it works, I'll make it identical to the C# logic (i.e. use `FindTokenOnLeftOfPosition`) but...
2. ...the C# code also uses the name span. That's not actually my addition; it's been in the product since the initial release, used to _disable_ the completion list.

The tests cover the cases of having no name and incomplete declarations for both languages - and I'll add a few more variations for good measure - so any future change to the behaviour should be caught.